nncf/quantization/telemetry_extractors.py

```diff
@@ -0,0 +1,23 @@
+"""
+ Copyright (c) 2022 Intel Corporation
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from typing import Any
+
+from nncf.telemetry.extractors import CollectedEvent
+from nncf.telemetry.extractors import TelemetryExtractor
+
+
+class CompressionStartedWithQuantizeApi(TelemetryExtractor):
+    def extract(self, _: Any) -> CollectedEvent:
+        return CollectedEvent(name="compression_started",
+                              data="quantize_api")
```

Comment:
I don't think that using `TelemetryExtractor` is the correct way to pass the event messages. This class should be used only to extract necessary information from an argument that has a difficult structure.


I have a question. Let's suppose we have 2 methods `func_1(`) and `func_2()` and these methods determine the different stages of our pipeline. These methods have a parameter which is called `some_param`. We decorate these methods and want to track this parameter. How do we understand where it comes from?

Reply:
You guessed it - you would have to set up custom extractors in both cases and differentiate the events by the event name. The event name is thus in general decoupled from both the function name and the argument name. I think that this decoupling is desirable because we can do refactoring on function names and argument names without having to submit conversion rules to whoever analyzes the telemetry. 

We are not tracking general parameter usage across functions, we are tracking specific events that give us valuable, analyzeable information. The alternative, as always, is to use direct `NNCFTelemetry.send_event` calls. If there is another idea that will allow us to flexibly collect events and their serializable data from function argument, please let me know.

Reply:
> The event name is thus in general decoupled from both the function name and the argument name. I think that this decoupling is desirable because we can do refactoring on function names and argument names without having to submit conversion rules to whoever analyzes the telemetry.

Yes, I also think so.

I have the following thoughts:

- Specify all events that we are interested in.

```python
class EventId(Enum):
     PTQ_API = 'PTQ_API'  # quatize(), quatize_impl()
     QAT_API = 'QAT_API '  # create_compressed_model()
     MODEL_EXPORT = 'MODEL_EXPORT'  # export_model()
     # Something else...
```

- Add `event_id` parameter to the `tracked_function` decorator.

```python
@tracked_function(
    NNCF_ONNX_CATEGORY,
    event_id=EventId.PTQ_API,
    extractors=['target_device', 'preset']
)
def quantize_impl(...):
    ...
```

- We send the following events:
    - `CollectedEvent(name=event_id.value)` - to identify the stage of the pipeline.
    - `CollectedEvent(name=f{event_id.value}:{argname})` - to track argument's value